server/src/main/java/org/elasticsearch/monitor/fs/FsHealthService.java

```diff
@@ -117,6 +118,8 @@ public StatusInfo getHealth() {
         Set<Path> unhealthyPaths = this.unhealthyPaths;
         if (enabled == false) {
             statusInfo = new StatusInfo(HEALTHY, "health check disabled");
+        } else if (brokenLock == true) {
+            statusInfo = new StatusInfo(UNHEALTHY, "health check failed on node due to broken locks");
```

Comment:
Minor wording nit: specify which lock was broken (and remove redundant `on node`), suggest this:

```suggestion
            statusInfo = new StatusInfo(UNHEALTHY, "health check failed due to broken node lock");
```